llvm/lib/Target/DirectX/DXILLegalizePass.cpp

```diff
@@ -143,6 +143,10 @@ static bool fixI8UseChain(Instruction &I,
     LoadInst *NewLoad = Builder.CreateLoad(ElementType, NewGEP);
     ReplacedValues[Load] = NewLoad;
     Load->replaceAllUsesWith(NewLoad);
+    Load->setOperand(Load->getPointerOperandIndex(),
+                     PoisonValue::get(CE->getType()));
```

Comment:
So we are replacing the  constExpr with a poison so we can delete it  since we have a delayed delete because of `ToRemove`. I think a comment to this effect would be good, because on first glance people are going to see this and scartch their heads.